node/src/Unison/Runtime/ResourcePool.hs

```diff
@@ -36,8 +35,8 @@ addResourceToMap release pool p r wait maxPoolSize getThread poolSizeR queue = d
       >> STM.atomically (TQ.writeTQueue queue key)
     else release r
 
-_acquire :: (Ord p, H.Hashable p) => (p -> IO r) -> (r -> IO ()) -> Cache p r -> MaxPoolSize -> IORef.IORef Int -> ReaperQueue p -> IO CC.ThreadId -> p -> Int -> IO (r, IO ())
-_acquire acquire release pool maxPoolSize poolSizeR queue getThread p wait = do
+_acquire :: (Ord p, H.Hashable p) => (p -> IO r) -> (r -> IO ()) -> Cache p r -> Int -> MaxPoolSize -> IORef.IORef Int -> ReaperQueue p -> IO CC.ThreadId -> p -> IO (r, IO ())
+_acquire acquire release pool wait maxPoolSize poolSizeR queue getThread p = do
   threadId <- getThread
   m <- M.lookup (p,threadId) pool
```

Comment:
@pchiusano That makes sense, but I cannot figure out what should happen if T1 acquires, then T2 attempts to require. Should T2 be blocked? Or acquire it without adding it to the pool? Without separating the key by threadId, if we add it to the cache, we cannot differentiate which one is which. Also, if T2 acquires and releases, then T1 releases, when T2 requests again, it would get the one from the pool, not the one it originally acquired and released. I am not sure if this is a problem or not. 

I could make it so that each new acquisition comes with a unique identifier used to keep track of "your" cached resource. Thoughts?